src/lib.rs

```diff
@@ -930,28 +908,7 @@ pub unsafe extern "C" fn dav1d_apply_grain(
                 .cast::<DRav1d<Rav1dITUTT35, Dav1dITUTT35>>())
             .update_rav1d();
         }
-        if let Some(mut seq_hdr_ref) = NonNull::new(out_c.seq_hdr_ref) {
-            (*seq_hdr_ref
-                .as_mut()
-                .data
-                .cast::<DRav1d<Rav1dSequenceHeader, Dav1dSequenceHeader>>())
-            .update_rav1d();
-        }
-        if let Some(mut frame_hdr_ref) = NonNull::new(out_c.frame_hdr_ref) {
-            (*frame_hdr_ref
-                .as_mut()
-                .data
-                .cast::<DRav1d<Rav1dFrameHeader, Dav1dFrameHeader>>())
-            .update_rav1d();
-        }
-        if let Some(mut itut_t35_ref) = NonNull::new(out_c.itut_t35_ref) {
-            (*itut_t35_ref
-                .as_mut()
-                .data
-                .cast::<DRav1d<Rav1dITUTT35, Dav1dITUTT35>>())
-            .update_rav1d();
-        }
-        let mut out_rust = out_c.into();
+        let mut out_rust = MaybeUninit::zeroed().assume_init(); // TODO(kkysen) Temporary until we return it directly.
```

Comment:
I assume the `update_ravid()` calls are unnecessary as we never read from `out_c`?

Reply:
Yeah exactly.